pkg/clustermesh/namespace/namespace.go

```diff
@@ -0,0 +1,100 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package namespace
+
+import (
+	"fmt"
+	"log/slog"
+	"strings"
+
+	"github.com/cilium/hive/cell"
+
+	"github.com/cilium/cilium/pkg/annotation"
+	"github.com/cilium/cilium/pkg/k8s/resource"
+	slim_corev1 "github.com/cilium/cilium/pkg/k8s/slim/k8s/api/core/v1"
+	"github.com/cilium/cilium/pkg/logging/logfields"
+)
+
+type managerParams struct {
+	cell.In
+
+	Logger     *slog.Logger
+	Config     Config
+	Namespaces resource.Resource[*slim_corev1.Namespace]
+	Lifecycle  cell.Lifecycle
+}
+
+type Manager interface {
+	IsGlobalNamespaceEnabledByDefault() bool
+	IsGlobalNamespaceByName(ns string) (bool, error)
+	IsGlobalNamespaceByObject(ns *slim_corev1.Namespace) bool
+}
+
+type manager struct {
+	logger     *slog.Logger
+	cfg        Config
+	namespaces resource.Resource[*slim_corev1.Namespace]
+	store      resource.Store[*slim_corev1.Namespace]
+}
+
+func newManager(params managerParams) *manager {
+	m := &manager{
+		logger:     params.Logger.With(logfields.LogSubsys, "clustermesh-namespace-manager"),
+		cfg:        params.Config,
+		namespaces: params.Namespaces,
+	}
+
+	params.Lifecycle.Append(cell.Hook{
+		OnStart: func(ctx cell.HookContext) error {
+			store, err := params.Namespaces.Store(ctx)
+			if err != nil {
+				return fmt.Errorf("failed to get namespace store: %w", err)
+			}
+			m.store = store
+			return nil
+		},
+	})
+
+	return m
+}
+
+// IsGlobalNamespaceEnabledByDefault returns whether the default global namespace
+// setting is enabled in the configuration.
+func (m *manager) IsGlobalNamespaceEnabledByDefault() bool {
+	return m.cfg.EnableDefaultGlobalNamespace
+}
+
+// IsGlobalNamespaceByObject determines whether the given namespace should be treated as a global
+// namespace based on its annotations and the provided configuration.
+func (m *manager) IsGlobalNamespaceByObject(ns *slim_corev1.Namespace) bool {
+	if ns == nil {
+		return false
+	}
+	// Get annotations map safely.
```

Comment:
What does this comment mean?

Reply:
Just meant that `GetAnnotation` is safe to use without taking any locks ourselves. Edited this comment to not be confusing.